src/ConstructiveSolidGeometry/LinePrimitives/LinePrimitives.jl

```diff
@@ -1,3 +1,16 @@
 struct LineSegments{T} <: AbstractLinePrimitive{T}
     points::Vector{CartesianPoint{T}}
 end
+
+function (*)(pl::LineSegments{T}, s::SVector{3,T})::LineSegments{T} where {T}
+    LineSegments(map(p -> p .* s, pl.points))
+end
+(*)(s::SVector{3,T}, pl::LineSegments{T}) where {T} = pl * s
+
+function (+)(pl::LineSegments{T}, t::CartesianVector{T})::LineSegments{T} where {T}
+    LineSegments(map(p -> p + t, pl.points))
+end
+
+function (*)(r::RotMatrix{3,T,TT}, pl::LineSegments{T})::LineSegments{T} where {T, TT}
+    LineSegments(map(p -> r * p, pl.points))
+end
```

Comment:
We should implement broadcasting (`.+`-syntax ) methods for our Point and Vector types. 
https://docs.julialang.org/en/v1/manual/arrays/#Broadcasting

Reply:
I did not come up with a fast way of implementing the broadcasting methods.

Broadcasting two arrays of different sizes (where one is treated as scalar) does not seem to be that simple as the broadcast method results in an error if the arrays are of different length.

One way to avoid treating a translation vector as `Array` of length 3 would be to pass it as an 1-dimensional `Tuple`:
```julia
@inline (+)(vp::Vector{CartesianPoint{T}}, v::CartesianVector{T}) where {T} = vp .+ (v,)
@inline (+)(vvp::Vector{Vector{CartesianPoint{T}}}, v::CartesianVector{T}) where {T} = vvp .+ (v,)  
vvp + v
```
However, this results in allocations which we do not want.

A loop gives allocation free result if the input vector is overwritten:
```julia
function add!(vvp::Vector{Vector{CartesianPoint{T}}}, v::CartesianVector{T}) where {T}
    for i in eachindex(vvp)
        vp::Vector{CartesianPoint{T}} = vvp[i]
        for j in eachindex(vp)
            vp[j] += v
        end
    end
    #vvp
end
add!(vvp, v)
```

@oschulz do you have a workaround for broadcasting differently sized arrays? 
Something in ArraysOfArrays.jl or UnsafeArrays.jl?

Reply:
Hmm, I also looked a bit into broadcasting but also did not find a good implementation. 

As a work around we can for sure define methods like `translate!`, `rotate!` and `scale!` for `Vector{CartesianPoint}`. 

Personally, I would like to have the `.+`-syntax as well. I would really appreciate input from @oschulz here as I am not sure whether it might not be a wrong usage of that syntax.

Reply:
I don't think we should wer our Point and Vector types as scalars under broadcasting. They are vectors, semantically, and so the default behavior for broadcast should be to broadcast over them. In applications where we do want to treat them as scalars (from a broadcast perspective), we'll just wrap them in a `Ref` or `Tuple`.

Reply:
So the Julianic syntax would be `vp .+ Ref(v)` or `vp .+ (v,)`.